--- ../rtt/os/oro_arch.h	2016-06-10 02:55:36.680061310 +0530
+++ ../patchstudy/oro_arch_study.h	2016-06-20 05:00:27.691200760 +0530
@@ -11,6 +11,8 @@
 #         include "oro_i386/oro_arch.h"
 #      elif defined( OROBLD_OS_ARCH_x86_64 )
 #         include "oro_x86_64/oro_arch.h"
+#      elif defined( OROBLD_OS_sparc)
+#         include "oro_sparc/oro_arch.h"
 #      elif defined( OROBLD_OS_ARCH_ppc )
 #         include "oro_powerpc/oro_arch.h"
 #      else
