diff -Nrpu -x build rock_base/tools/orocos.rb/ext/CMakeLists.txt rock_old/tools/orocos.rb/ext/CMakeLists.txt
--- rock_base/tools/orocos.rb/ext/CMakeLists.txt	2011-09-10 19:52:14.000000000 +0200
+++ rock_old/tools/orocos.rb/ext/CMakeLists.txt	2011-10-23 18:24:21.038742799 +0200
@@ -1,4 +1,8 @@
 cmake_minimum_required(VERSION 2.6)
+if($ENV{OROCOS_TARGET} STREQUAL "rtems")
+ MESSAGE(STATUS "Setting RTEMS enviroment")
+ set(CMAKE_TOOLCHAIN_FILE  rtems_toolchain.cmake)
+endif($ENV{OROCOS_TARGET} STREQUAL "rtems")
 PROJECT(corba)
 
 SET (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}")
@@ -118,5 +122,5 @@ if (RTT_MQUEUE_FOUND)
         ${RTT_PLUGIN_rtt-transport-mqueue_LIBRARY})
 endif()
 
-INSTALL(TARGETS rorocos_ext LIBRARY DESTINATION ".")
+INSTALL(TARGETS rorocos_ext LIBRARY DESTINATION "." ARCHIVE DESTINATION ".")
 
Los archivos binarios rock_base/tools/orocos.rb/ext/rorocos_ext.a y rock_old/tools/orocos.rb/ext/rorocos_ext.a son distintos
Los archivos binarios rock_base/tools/orocos.rb/ext/rorocos_ext.so y rock_old/tools/orocos.rb/ext/rorocos_ext.so son distintos
diff -Nrpu -x build rock_base/tools/orocos.rb/ext/rtems_toolchain.cmake rock_old/tools/orocos.rb/ext/rtems_toolchain.cmake
--- rock_base/tools/orocos.rb/ext/rtems_toolchain.cmake	1970-01-01 01:00:00.000000000 +0100
+++ rock_old/tools/orocos.rb/ext/rtems_toolchain.cmake	2011-10-23 02:57:37.737976176 +0200
@@ -0,0 +1,85 @@
+INCLUDE(CMakeForceCompiler)
+
+SET(INSTALL_PREFIX $ENV{INSTALL_PREFIX})
+SET(TARGET $ENV{TARGET})
+SET(BSP $ENV{BSP})
+
+SET(CMAKE_SYSTEM_NAME RTEMS)
+
+CMAKE_FORCE_C_COMPILER($ENV{TARGET}-gcc GNU)
+CMAKE_FORCE_CXX_COMPILER($ENV{TARGET}-g++ GNU)
+
+SET(__BSD_VISIBLE 1)
+
+set(CFLAGS "-B${INSTALL_PREFIX}/rtems/${TARGET}/${BSP}/lib/ -specs bsp_specs -qrtems  -D__RTEMS__ -DBoost_USE_STATIC_LIBS -D__BSD_VISIBLE -DBOOST_SYSTEM_NO_DEPRECATED   -DBOOST_DISABLE_THREADS  -DRUBY_EXTENSIONS_AVAILABLE=NO")
+
+IF ("${TARGET}" MATCHES "i?86-*")
+       set (CFLAGS "-g -O2 -march=i486 -mtune=i486 -D__i386__ -Wl,-Ttext,0x00100000 ${CFLAGS}")
+ELSEIF ("${TARGET}" MATCHES "sparc-*")
+       set (CFLAGS "${CFLAGS} -D__SPARC__")
+ENDIF()
+
+SET (CXXFLAGS "${CFLAGS} -fpermissive -fexceptions")
+
+add_definitions(${CXXFLAGS})
+
+SET (CMAKE_C_FLAGS ${CFLAGS})
+SET (CMAKE_CXX_FLAGS ${CXXFLAGS})
+
+SET(CMAKE_CROSSCOMPILING ON)
+
+SET(BOOST_ROOT $ENV{INSTALL_PREFIX}/boost/)
+
+SET(BOOST_INCLUDEDIR $ENV{INSTALL_PREFIX}boost/include)
+SET(OMNIORB_INCLUDEDIR  $ENV{INSTALL_PREFIX}/omniORB/include)
+SET(EIGEN2_INCLUDE_DIR $ENV{INSTALL_PREFIX}/eigen/include)
+
+SET(BOOST_LIBRARYDIR $ENV{INSTALL_PREFIX}boost/lib)
+SET(OMNIORB_LIBRARYDIR  $ENV{INSTALL_PREFIX}/omniORB/lib)
+SET(EIGEN2_LIBRARYDIR $ENV{INSTALL_PREFIX}/eigen/lib)
+
+SET(BOOST_LIBRARYDIR $ENV{INSTALL_PREFIX}boost/lib)
+
+SET(Boost_USE_STATIC_LIBS ON)
+SET(Boost_ADDITIONAL_VERSIONS "1.44.0" "1.44")
+SET(Boost_COMPILER -rtems4.10)
+
+#TODO: Clean  link and include directories
+include_directories(${BOOST_INCLUDEDIR} ${OMNIORB_INCLUDEDIR} ${EIGEN2_INCLUDE_DIR})
+link_directories(${BOOST_LIBRARYDIR} ${OMNIORB_LIBRARYDIR} ${EIGEN2_LIBRARYDIR})
+
+SET(DCMAKE_SYSTEM_NAME RTEMS)
+
+SET(CMAKE_SHARED_LIBRARY_C_FLAGS "")              # -pic 
+SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "")       # -shared
+SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "")         # +s, flag for exe link to use shared lib
+SET(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG "")       # -rpath
+SET(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG_SEP "")   # : or empty
+
+SET(CMAKE_LINK_LIBRARY_SUFFIX "")
+SET(CMAKE_STATIC_LIBRARY_PREFIX "lib")
+SET(CMAKE_STATIC_LIBRARY_SUFFIX ".a")
+SET(CMAKE_SHARED_LIBRARY_PREFIX "lib")          # lib
+SET(CMAKE_SHARED_LIBRARY_SUFFIX ".a")           # .a
+SET(CMAKE_EXECUTABLE_SUFFIX ".bin")             #
+SET(CMAKE_DL_LIBS "" )
+
+SET(CMAKE_FIND_LIBRARY_PREFIXES "lib")
+SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+
+# RTEMS doesn't support shared libs
+SET_PROPERTY(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS FALSE)
+SET(CMAKE_CXX_LINK_SHARED_LIBRARY )
+SET(CMAKE_CXX_LINK_MODULE_LIBRARY )
+SET(CMAKE_C_LINK_SHARED_LIBRARY )
+SET(CMAKE_C_LINK_MODULE_LIBRARY )
+
+# set flags 
+message("-- Setting static RTEMS")
+SET(BUILD_STATIC ON)
+
+message("-- Not building shared")
+SET(NOT_BUILD_SHARED ON)
+
+#Include rtems directories and libraries
+include_directories (${INSTALL_PREFIX}/rtems/${TARGET}/${BSP}/lib/include ${INSTALL_PREFIX}/rtems/${TARGET}/include ${INSTALL_PREFIX}${TARGET}/${BSP}/lib/include/networking)
diff -Nrpu -x build rock_base/tools/orocos.rb/ext/RubyExtensions.cmake rock_old/tools/orocos.rb/ext/RubyExtensions.cmake
--- rock_base/tools/orocos.rb/ext/RubyExtensions.cmake	2011-09-10 19:52:14.000000000 +0200
+++ rock_old/tools/orocos.rb/ext/RubyExtensions.cmake	2011-10-23 18:23:23.466371605 +0200
@@ -30,9 +30,13 @@ ELSE(NOT RUBY_INCLUDE_PATH)
 	LINK_DIRECTORIES(${rubylib_path})
 
 	SET_SOURCE_FILES_PROPERTIES(${ARGN} PROPERTIES COMPILE_FLAGS "${RUBY_CFLAGS}")
-	ADD_LIBRARY(${target} MODULE ${ARGN})
-        set_target_properties(${target} PROPERTIES
-            LINK_FLAGS "-z noexecstack")
+   IF (NOT $ENV{OROCOS_TARGET} STREQUAL "rtems")
+	   ADD_LIBRARY(${target} MODULE ${ARGN})
+      set_target_properties(${target} PROPERTIES
+              LINK_FLAGS "-z noexecstack")
+   ELSE()
+	   ADD_LIBRARY(${target} STATIC ${ARGN})
+   ENDIF()
 	SET_TARGET_PROPERTIES(${target} PROPERTIES PREFIX "")
     ENDMACRO(ADD_RUBY_EXTENSION)
 ENDIF(NOT RUBY_INCLUDE_PATH)

