diff -rupN newlib-1.18.0.orig/newlib/libc/sys/rtems/sys/dirent.h newlib-1.18.0/newlib/libc/sys/rtems/sys/dirent.h
--- newlib-1.18.0.orig/newlib/libc/sys/rtems/sys/dirent.h	2004-01-08 20:25:21.000000000 +0100
+++ newlib-1.18.0/newlib/libc/sys/rtems/sys/dirent.h	2011-10-01 15:59:18.765780591 +0200
@@ -26,6 +26,7 @@ typedef struct _dirdesc {
 
 DIR *opendir(const char *);
 struct dirent *readdir(DIR *);
+int readdir_r(DIR *dirp, struct dirent *entry, struct dirent **result);
 void rewinddir(DIR *);
 int closedir(DIR *);
 void seekdir(DIR *dir, long loc);
